tools/trace.py

```diff
@@ -630,8 +630,8 @@ def print_event(self, bpf, cpu, data, size):
                 event = ct.cast(data, ct.POINTER(self.python_struct)).contents
                 if self.name not in event.comm:
                     return
-                values = map(lambda i: getattr(event, "v%d" % i),
-                             range(0, len(self.values)))
+                values = [getattr(event, "v%d" % i).decode()
```

Comment:
Do we need 'decode()' here? With this patch, I tried locally and got the following error with python3.6.
```
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 234, in 'calling callback function'
  File "/usr/lib/python3.6/site-packages/bcc/table.py", line 982, in raw_cb_
    callback(cpu, data, size)
  File "./trace.py", line 634, in print_event
    for i in range(len(self.values))]
  File "./trace.py", line 634, in <listcomp>
    for i in range(len(self.values))]
AttributeError: 'int' object has no attribute 'decode'
```

The original code doesn't have 'decode()'. should we add decode() only for string field of event?

Reply:
Thank you for catching it! It was something else that I was trying and indeed will only work with strings. This `int/const char*` duality is present in many places of the code base, not sure if it should be addressed in a general manner at the end of the day.

For users who only go for the tools without Python knowledge, they might be surprised to get a series of `b'something` when printing results. 

What do you think of:
```python
                values = [value.decode()
                          if hasattr(value, "decode") else value
                          for value in
                           (getattr(event, "v%d" % i)
                            for i in range(len(self.values)))]
```

Reply:
The above is better. Let us first fix trace.py. We can generalize it if we have reports from other tools. Thanks.